### PR TITLE
Fix opengl 3.2 ver

### PR DESCRIPTION
**Pull request by vladus7000**

### CODE REVIEW

File: src/game/main.cpp [hunk 1]
```diff
@@ -20,8 +20,14 @@ int main(int _argc, char** _argv)
     SDL_Window *window;        // Declare a pointer to an SDL_Window
     
 	SDL_GL_SetAttribute(SDL_GL_ACCELERATED_VISUAL, 1);
+
+#ifdef FB_WIN
```
Comment (KVADRO) [inline src/game/main.cpp]: We need to use 3.3 on all platforms, please remove this branching

File: src/engine/render/opengl_renderer/OpenGLRenderer.cpp [hunk 2]
```diff
@@ -1,8 +1,8 @@
 #include "OpenGLRenderer.hpp"
-#ifdef _WIN32
-#include "gl_core_4_4_mac.h"
-#else
+#ifdef FB_WIN
```
Comment (KVADRO) [inline src/engine/render/opengl_renderer/OpenGLRenderer.cpp]: We need to use 3.3 on all platforms, please remove this branching

File: src/engine/CMakeLists.txt [hunk 3]
```diff
@@ -9,7 +9,7 @@ file(GLOB_RECURSE RENDER_SOURCES render/*.cpp render/*.hpp)
 if(WIN32)
 set (GL_CORE_SRCS ${CMAKE_SOURCE_DIR}/src/engine/render/opengl_renderer/gl_core_4_4.c)
```
Comment (KVADRO) [inline src/engine/CMakeLists.txt]: We need to use 3.3 on all platforms, please remove this branching

**Comment by vladus7000** [comment]
So ?

**Review by KVADRO — CHANGES_REQUESTED** [review]

File: src/engine/CMakeLists.txt (hunk 3 above)
Comment (vladus7000) [reply src/engine/CMakeLists.txt]: why ? we can support 4/3 easy

Comment (KVADRO) [reply src/engine/CMakeLists.txt]: don't need to

Comment (KVADRO) [reply src/engine/CMakeLists.txt]: with this you create fragmentation without no need. Do you need features specific to version 4.4?

Comment (vladus7000) [reply src/engine/CMakeLists.txt]: with this change we will limit users from getting new openGL features

Comment (vladus7000) [reply src/engine/CMakeLists.txt]: OpenGL major version numbers effectively represent hardware levels, like D3D major version numbers. GL 2.x represents DX9 hardware. GL 3.x represents DX10-class hardware. And therefore GL 4.x represents DX11-class hardware.

Not all features introduced in GL 4.x are hardware-specific, however. Some are API convenience and cleanup features. These are available as extensions to lower-level hardware, and drivers from NVIDIA and AMD do provide them. The biggest of these is program separation, which is something GLSL has needed for a long time.

In terms of actual hardware features presented by GL 4.x that are not available in 3.x, it isn't huge or game-changing. It's certainly significant, but not earth-moving. You can find a detailed breakdown of GL features by version on the OpenGL Wiki.

Image load/store and atomic counters are probably the biggest ones (this includes shader storage buffers too), though simultaneously the least advertised. Shaders get to write/read from images and buffers directly, though there are a lot of caveats and synchronization issues to understand about doing this.

GL 4.3 added compute shaders, which coupled with the above allow you to do computational tasks without the pain of OpenCL/OpenGL interop.

The feature that gets talked about the most is tessellation. This introduces an additional pair of shader stages: one that decides how much to tessellate a primitive, and the other to decide how to generate the new values from the tessellated primitive.

Subroutines are an interesting but often overlooked feature. Basically, you can attach subroutines to shader programs like uniforms. So you could use the same basic program that will run a lighting function, but you can swap lighting functions without changing the program. The basic program would fetch textures, figure out what the diffuse/specular/etc material parameters are, determine the normal and lighting parameters, and pass that to the subroutine. You could have a wide variety of lighting subroutines, from the quick and simple Lambertian diffuse to full-on Cook-Torrance (though you'd need some more material parameters).

Comment (KVADRO) [reply src/engine/CMakeLists.txt]: If you can list features that really needed for this project and this features provided only in 4.4, please specify them. 
1) With 3.3 we can target broad range of old hardware.
2) As i told, osx 'stable' opengl version is 3.3, even if you'll have 4.4 on win, you will not be able to use functionality from 4.4 without creating even more fragmentation
3) Using one opengl version we can omit not necessary branching #ifdef WIN ... #else ...  that will be spreaded through whole project

**Review by KVADRO — APPROVED** [review]

**Pull request merged**